File: backend/app/services/honorific.py
"""氏名の敬称ユーティリティ。

フロント `lib/honorific.ts`(withHonorific/hasHonorific)と同一規則。
既に敬称(様/さま/さん)で終わる名前(デモの来訪者名「審査員様」、法人名+様、"お客様" 等)
には「様」を重ねず、それ以外の氏名には「様」を付ける。キオスク/管理画面/メール/通知
すべてで訪問者名の敬称表示をこの規則に統一する。
"""

_HONORIFICS = ("様", "さま", "さん")


def has_honorific(name: str | None) -> bool:
    """名前が既に敬称(様/さま/さん)で終わっているか。"""
    return (name or "").strip().endswith(_HONORIFICS)


def with_honorific(name: str | None) -> str:
    """氏名に「様」を付ける。既に敬称で終わる名前には重ねない。空文字はそのまま返す。"""
    n = (name or "").strip()
    if not n:
        return n
    return n if n.endswith(_HONORIFICS) else f"{n} 様"

File: frontend/lib/honorific.ts
// 氏名への敬称「様」付与ユーティリティ。
// デモの来訪者名「審査員様」等、既に敬称で終わる名前に「様」を重ねて
// 「審査員様 様」になる二重敬称を防ぐ。バックエンド _with_honorific と同じ規則。

const HONORIFIC_RE = /(?:様|さま|さん)$/;

/** 名前が既に敬称(様/さま/さん)で終わっているか。 */
export function hasHonorific(name: string | null | undefined): boolean {
  return HONORIFIC_RE.test((name ?? "").trim());
}

/** 氏名に「様」を付ける。既に敬称で終わる名前には重ねない。 */
export function withHonorific(name: string | null | undefined): string {
  const n = (name ?? "").trim();
  if (!n) return n;
  return hasHonorific(n) ? n : `${n} 様`;
}
